app/javascript/components/lease_ups/LeaseUpsTableContainer.js

```diff
@@ -81,13 +81,15 @@ class LeaseUpTableContainer extends React.Component {
       const response = await apiService.createLeaseUpStatus(data)
 
       if (response) {
-        // find the application in question in the applications table data
-        // and update its lease up status value and status updated date
+        // find the rows with the application id whos status is being updated
+        // and update their lease up status value and status updated date
         // in the table
-        const applicationIndex = findIndex(this.state.applications, { id: applicationPreferenceId })
-
-        this.updateResults(`[${applicationIndex}]['lease_up_status']`, status)
-        this.updateResults(`[${applicationIndex}]['status_updated']`, moment().format(utils.SALESFORCE_DATE_FORMAT))
+        each(this.state.applications, (app, index) => {
```

Comment:
this is to update for all visible rows that have the same application, right? Does the status get updated in salesforce too? Is it possible that there would be an application/preference combo that isn't part of the state that would need to be updated?

I might just need some more info about what this is.

Reply:
`this is to update for all visible rows that have the same application, right?` -> Yes
` Does the status get updated in salesforce too?` -> Yes `const response = await apiService.createLeaseUpStatus(data)` does the salesforce request before we do the ui change.
`Is it possible that there would be an application/preference combo that isn't part of the state that would need to be updated?` -> From a UI standpoint, no because we store all the data in state. In terms of the field update comment, we just create a new one and assign it to the application once and that comment will be the same status for all preference rows because it's the same application :)

Reply:
So the business rule here is that an application can be in multiple preference buckets so it can have a preference rank for multiple preferences. It still shares the same status coz it's the same application, but if I'm paginating from COP to the DTHP ranks in this table, and the application has a COP rank and also has DTHP rank, I want to see the application row for each with the correct status. And when I update the status, I want that to reflect on all the rows correctly as well.